linera-service/src/database_tool.rs

```diff
@@ -0,0 +1,186 @@
+    #[structopt(name = "initialize")]
+    Initialize {
+        /// Storage configuration for the blockchain history.
+        #[structopt(long = "storage")]
+        storage_config: String,
+
+        /// The maximal number of simultaneous queries to the database
+        #[structopt(long)]
+        max_concurrent_queries: Option<usize>,
+
+        /// The maximal number of simultaneous stream queries to the database
+        #[structopt(long, default_value = "10")]
+        max_stream_queries: usize,
+
+        /// The maximal number of entries in the storage cache.
+        #[structopt(long, default_value = "1000")]
+        cache_size: usize,
+    },
+}
+
+#[tokio::main]
+async fn main() -> Result<(), anyhow::Error> {
+    let options = DatabaseToolOptions::from_args();
+
+    match options.command {
+        DatabaseToolCommand::DeleteAll {
+            storage_config,
+            max_concurrent_queries,
+            max_stream_queries,
+            cache_size,
+        } => {
+            let storage_config: StorageConfig = storage_config.parse()?;
+            let common_config = CommonStoreConfig {
+                max_concurrent_queries,
+                max_stream_queries,
+                cache_size,
+            };
+            let full_storage_config = storage_config.add_common_config(common_config).await?;
+            full_storage_config
+                .delete_all()
+                .await
+                .expect("successful delete_all operation");
+        }
+        DatabaseToolCommand::DeleteSingle {
+            storage_config,
+            max_concurrent_queries,
+            max_stream_queries,
+            cache_size,
+        } => {
+            let storage_config: StorageConfig = storage_config.parse()?;
+            let common_config = CommonStoreConfig {
+                max_concurrent_queries,
+                max_stream_queries,
+                cache_size,
+            };
+            let full_storage_config = storage_config.add_common_config(common_config).await?;
+            full_storage_config
+                .delete_single()
+                .await
+                .expect("successful delete_all operation");
+        }
+        DatabaseToolCommand::Existence {
+            storage_config,
+            max_concurrent_queries,
+            max_stream_queries,
+            cache_size,
+        } => {
+            let storage_config: StorageConfig = storage_config.parse()?;
+            let common_config = CommonStoreConfig {
+                max_concurrent_queries,
+                max_stream_queries,
+                cache_size,
+            };
+            let full_storage_config = storage_config.add_common_config(common_config).await?;
+            let test = full_storage_config
+                .test_existence()
+                .await
+                .expect("successful delete_all operation");
+            if test {
+                println!("The database does exist");
```

Comment:
We're trying to print to stdout only parsable data. So here `tracing::error!` is recommended

Reply:
Ok.